tests/Backend/ExternalBuckets/SnowflakeSchemaToRegisteredBucketTest.php

```diff
@@ -0,0 +1,342 @@
+            $stage,
+            'will not fail',
+            'snowflake',
+            $bucketName,
+        );
+
+        $tables = $this->_client->listTables($stage.'.'.$bucketName);
+        $this->assertCount(1, $tables);
+
+        $shareToken = $this->linkingClient->verifyToken();
+        $targetProjectId = $shareToken['owner']['id'];
+
+        $this->shareClient->shareBucketToProjects($registeredBucketId, [$targetProjectId]);
+
+        $sharedBucket = $this->_client->getBucket($registeredBucketId);
+        $this->assertEquals('specific-projects', $sharedBucket['sharing']);
+        $this->assertEquals($targetProjectId, $sharedBucket['sharingParameters']['projects'][0]['id']);
+
+        // LINKING START
+
+        $token = $this->_client->verifyToken();
+        $linkedBucketId = $this->linkingClient->linkBucket('LINKED_BUCKET', 'in', $token['owner']['id'], $sharedBucket['id'], 'LINKED_BUCKET');
+        $linkedBucket = $this->linkingClient->getBucket($linkedBucketId);
+        $this->assertEquals($sharedBucket['id'], $linkedBucket['sourceBucket']['id']);
+
+        $linkingTables = $this->linkingClient->listTables($linkedBucketId);
+        $this->assertCount(1, $tables);
+        $linkingTable = $linkingTables[0];
+
+        $dataPreview = $this->linkingClient->getTableDataPreview($linkingTable['id']);
+        $this->assertEquals(
+            <<<EXPECTED
+"ID","LASTNAME"
+"1","Novák"
+
+EXPECTED,
+            $dataPreview,
+        );
+
+        $linkingWorkspaces = new Workspaces($this->linkingClient);
+        $linkingWorkspace = $linkingWorkspaces->createWorkspace([], true);
+        $linkingBackend = WorkspaceBackendFactory::createWorkspaceBackend($linkingWorkspace);
+
+        /** @var \Keboola\Db\Import\Snowflake\Connection $linkingSnowflakeDb */
+        $linkingSnowflakeDb = $linkingBackend->getDb();
+
+        $result = $linkingSnowflakeDb->fetchAll(sprintf(
+            'SELECT * FROM %s.%s.%s',
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_DB),
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_SCHEMA),
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_TABLE),
+        ));
+        $this->assertEquals(
+            [
+                [
+                    'ID' => 1,
+                    'LASTNAME' => 'Novák',
+                ],
+            ],
+            $result,
+        );
+
+        $options = [
+            'input' => [
+                [
+                    'source' => $linkingTable['id'],
+                    'destination' => 'languages',
+                ]
+            ],
+        ];
+
+        $options = InputMappingConverter::convertInputColumnsTypesForBackend(
+            $linkingWorkspace['connection']['backend'],
+            $options,
+        );
+
+        $linkingWorkspaces->loadWorkspaceData($linkingWorkspace['id'], $options);
+        $linkingWorkspaces->cloneIntoWorkspace($linkingWorkspace['id'], $options);
```

Comment:
Proč mi sem commituješ tenhle WIP?

Reply:
Jo sry, schvalne aby som ukazal qko som to myslel, pišem o tom v komentari tak aby som na to mohol odkazat.

Reply:
Ale vždyť o tomhle loadu jsme na SU říkali, že jej vyřešíme bokem a jen musíme ověřit, že je zakázaný: https://keboola.atlassian.net/browse/CT-1685

Reply:
Jee mna mate jak je to roztazkovane. A ze nieco nefunguje ale neni to osetrene. Tak ja dropnem ten wip. Ale pobavil by som sa este o tom preco to nejde. Resp. Mne pride ze by to mohlo fungovat tak by bolo mozno fajn to niekam popisat preco to nejde qlebo preco to nechceme to bude asi na @jirkasemmler

Reply:
já si ten WIP dropnu

Reply:
Bol som rychlejsi :D

Reply:
jo a já to už podruhé rebasoval …

Reply:
Sorry 🫠